src/util.h

```diff
@@ -18,6 +19,18 @@
 
 namespace node {
 
+#ifdef __GNUC__
+#define NO_RETURN __attribute__((noreturn))
+#else
+#define NO_RETURN
+#endif
+
+// The slightly odd function signature for Assert() is to
```

Comment:
Yes.  I'll reword the comment to not mention 'register pressure' because it's really about instruction cache pressure.

With a normal signature, the core of a CHECK is 7 instructions on x64 with `-mcmodel=small`.  This approach brings it down to 4 instructions.

It probably doesn't matter too much in the grand scheme of things but I figured that, since it's hidden behind a macro, I might as well go for maximum efficiency.